### PR TITLE
Add filtering capabilities (only by name)

### PR DESCRIPTION
**Pull request by carlos-chourio**
Correct typo in "seperate"
Add Extension Method to add comparison mode to Contains method in string type.
Improve PropertyDidChange method to obtain the name of the property without specifying the name directly.

### CODE REVIEW

File: VersionChanger/Data/NotifyableObject.cs [hunk 1]
```diff
@@ -3,11 +3,9 @@
 using System.Collections.Generic;
 using System.ComponentModel;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
 
-namespace DSoft.VersionChanger.Data
-{
+namespace DSoft.VersionChanger.Data {
```
Comment (newky2k) [inline VersionChanger/Data/NotifyableObject.cs:8]: The { should be on a new line

**Review by newky2k — CHANGES_REQUESTED** [review]
Hi,

Thanks for you contribution.  I will take a wider look at this later this week and then look at merging and publishing a new version.

Just one fix to the PR to do with formattting

**Review by newky2k — APPROVED** [review]

**Comment by carlos-chourio** [comment]
Hi Dave, I saw you approved my changes, that's great since it's my first contribution to an open source project. I'd like to make a few other changes to code, first: I'd like to change the click events on the buttons to commands (cus that's the best practice when using mvvm).
I'd also like to change the functionality of the *select all checkbox* when using the search box to select only the filtered items and not all of them. 
What do you think?

**Comment by newky2k** [comment]
hi @carlos-chourio,

Firstly let me say thanks for your contribution.

Thats fine with me, but you'll need to add some event for the window to subscribe too so that it can be told to close once the update has completed, which is probably why i did it as a click event at the time.  

We should also make sure the addin only processes the items that have been filtered AND have been checked.

You'll also need to move this into the Fields region, just to keep the code in order

private ICollectionView _ItemsView { get; set; }

But again thanks for you help, this project has been ongoing for a number of years and has evolved as my understanding has evolved so it can be a bit schizophrenic.

Thanks
Dave

**Comment by carlos-chourio** [comment]
Hello Dave, I haven't had enough time to work on the features I told you, however i'd like to start working on them soon. And I want to ask you something before I start, do you think it's possible to use the prism library in an vs extension? I'd like to use their RelayCommand implementation, nevertheless if not possible I could use my own.

**Comment by newky2k** [comment]
@carlos-chourio  adding the prism library is probably a bit heavy just to implement commands on the buttons tbh, as we're not really creating any super modular.  Happy for you to use your own, i've used this for all of my MVVM projects in the past -> https://github.com/newky2k/System.Mvvm/blob/master/System.Mvvm/DelegateCommand.cs

**Comment by newky2k** [comment]
@carlos-chourio  Also you need to make sure the command calls ThreadHelper.ThrowIfNotOnUi
Thread -> https://github.com/newky2k/VersionChangerAddin/blob/master/VersionChanger/Views/VersionChanger.xaml.cs#L42 as this is a requirement now in Visual Studio when calling code that might update the UI

**Pull request merged**